drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenter.java

```diff
@@ -217,15 +230,16 @@ public void reloadTestTools(boolean disable) {
     }
 
     public void onRunScenario() {
-        List<Integer> indexes = IntStream.range(0, context.getStatus().getSimulation().getUnmodifiableScenarios().size())
+        List<Integer> indexes = IntStream.range(0, focusedContext.getStatus().getSimulation().getUnmodifiableScenarios().size())
                 .boxed()
                 .collect(Collectors.toList());
         onRunScenario(indexes);
     }
 
     public void onRunScenario(List<Integer> indexOfScenarioToRun) {
-        view.getScenarioGridPanel().getScenarioGrid().getModel().resetErrors();
-        model.setSimulation(context.getStatus().getSimulation());
+        scenarioMainGridWidget.resetErrors();
+        scenarioBackgroundGridWidget.resetErrors();
```

Comment:
@gitgabrio Well, in this case it could be reasonable, if the future background table will support error handling in the same way of the original one. I mean, I can remove it, but I pretty sure it will required in the next steps. Wdyt?

Reply:
Removed.